### ISSUE TITLE
Performance profile "cpu counters" ignores subprocesses?

### ISSUE THREAD

**Issue by philwo** [bazelbuild/bazel#10657]
### Description of the problem / feature request:

The "cpu counters" measurement doesn't seem right. When I look at `htop` on a CI machine that is running a Bazel postsubmit, basically all cores are 100% all of the time. However, in the JSON performance profile, the "cpu counters" graph looks like this:

<img width="628" alt="Screenshot 2020-01-26 at 18 53 02" src="https://user-images.githubusercontent.com/504652/73139310-166bd500-406d-11ea-9893-c8e38bad16ec.png">

Maybe it is only measuring the CPU consumption of Bazel's own process, not of the entire build? This seems weird, as the CPU consumption of Bazel itself is probably only of concern to Bazel developers and not actionable for users who want to find out why their build was slow. If we had CPU measurements for the entire build (which, if too hard, can probably be well approximated by just looking at the entire system, i.e. load average or the cores how htop measures them), then we could see from the profile if a build is slow because it's CPU bound or not.

### What's the output of `bazel info release`?

Bazel 2.0.0

**Comment by meisterT** [comment]
The CPU counters indeed measure only Bazel's own CPU usage, intentionally so: this reflects the overhead that Bazel is adding to the build. We were also biased by the fact that we run most actions via remote execution at Google.

It's non-trivial to get "live data" back from the subprocesses that we spawned (having to make this work on multiple OSs doesn't help either). So one way to help you out is here to provide an aggregated view over the ExecutionStatistics that we collect.

**Comment by seanxwzhang** [comment]
@meisterT just a quick additional follow-up, bazel seems to provide a `process-wrapper` that's able to track spawned process's resource usage by using linux's `wait4` system call (https://github.com/bazelbuild/bazel/blob/master/src/main/tools/process-tools.cc#L195). It also seems that the resource usage stats are dumped in a temporary directory https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/exec/local/LocalSpawnRunner.java#L364.

I was trying to look for such `stats.out` under `execRoot` but no avail, where does the stats.out file live? 

fwiw, by invoking `bazel test -s //:my_python_test`, I got 
```
external/bazel_tools/tools/test/test-setup.sh my_python_test
```
, where `my_python_test` is constructed from the [python stub template file ](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/bazel/rules/python/python_stub_template.txt). So I guess the process wrapper wasn't used at all? 

Thanks!

**Comment by benjaminp** [comment]
You have to pass `--experimental_collect_local_action_metrics`.

**Comment by seanxwzhang** [comment]
thanks @benjaminp, I did passed `--experimental_collect_local_action_metrics` and I still wasn't able to find the `stats.out` file (or even an temporary named `local-spawn-runner.` under my execroot, as suggested in https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/exec/local/LocalSpawnRunner.java#L189).

Perhaps I missed something?

**Comment by benjaminp** [comment]
Yes, the local implementation leaves something to be desired in that it puts the stats proto in a temporary directory that's unconditionally deleted. At least with sandbox execution you can pass `--experimental_collect_local_action_metrics` and `--sandbox_debug` and poke around in the left-over sandboox root to get `stats.out`.

**Comment by seanxwzhang** [comment]
> Yes, the local implementation leaves something to be desired in that it puts the stats proto in a temporary directory that's unconditionally deleted. At least with sandbox execution you can pass `--experimental_collect_local_action_metrics` and `--sandbox_debug` and poke around in the left-over sandboox root to get `stats.out`.

OIC, yeah I saw it was deleted after execution. FWIW, I was trying to get the complete resource usage stats from executions since the local implementation [only includes a subset](https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/exec/local/LocalSpawnRunner.java#L433-L440) of what would be available from [getrusage](http://man7.org/linux/man-pages/man2/getrusage.2.html). (the existing statistics missed memory stats, which is what I'm trying to get)

I was hoping that I could avoid reinventing the wheel but it seems it's inevitable. Forgive my bazel-ignorance but given the extent to which a simple max_rss metric could help people debug a memory-leaking test, why was memory stats (max_rss) not part of SpawnResult (from local execution, that is)?

**Comment by meisterT** [comment]
To show how it looks after the most recent commit:
![image](https://user-images.githubusercontent.com/418721/133793385-afeb1375-969d-4f2b-a6f5-5b2a3e1a8781.png)

You can clearly see that the total system CPU usage spiked once Bazel started executing actions.
